rules/linux/persistence_init_d_file_creation.toml

```
@@ -0,0 +1,56 @@
+[metadata]
+creation_date = "2023/03/21"
+integration = ["endpoint"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2023/04/03"
+
+[rule]
+author = ["Elastic"]
+description = """
+Files that are placed in the /etc/init.d/ directory in Unix can be used to start custom applications,
+services, scripts or commands during start-up. Init.d has been mostly replaced in favor of Systemd, however, 
+through the "systemd-sysv-generator" init.d files can be converted to service unit files that run at boot.
+Adversaries may add or alter files located in the /etc/init.d/ directory to execute malicious code on boot
+time in order to gain persistence onto the system. 
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*", "endgame-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential Persistence Through init.d Detected"
+references = [
+    "https://www.intezer.com/blog/malware-analysis/hiddenwasp-malware-targeting-linux-systems/",
+    "https://pberba.github.io/security/2022/02/06/linux-threat-hunting-for-persistence-initialization-scripts-and-shell-configuration/#8-boot-or-logon-initialization-scripts-rc-scripts",
+    "https://www.cyberciti.biz/faq/how-to-enable-rc-local-shell-script-on-systemd-while-booting-linux-system/"
+
+]
+risk_score = 47
+rule_id = "474fd20e-14cc-49c5-8160-d9ab4ba16c8b"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Persistence", "Elastic Endgame"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+file where host.os.type == "linux" and 
+event.action in ("creation", "file_create_event", "rename", "file_rename_event") and 
+process.name : ("*sh", "nano", "vi*", "mv", "cp", "touch", "editor", "sed") and file.path : "/etc/init.d/*" and not 
+endsWith(file.name, "swp") and user.id == "0"
```

```suggestion
host.os.type :"linux" and event.action:("creation" or "file_create_event" or "rename" or "file_rename_event") and 
file.path : "/etc/init.d/*" and not process.executable : ("/usr/bin/dpkg" or "/usr/bin/dockerd" or "/bin/rpm")
```

you can also use the existing rule telem to exclude more known noisy process.executable values.